component/org.wso2.carbon.identity.oauth2.validators.xacml/src/main/java/org/wso2/carbon/identity/oauth2/validators/xacml/XACMLScopeValidator.java

```diff
@@ -149,14 +157,71 @@ private RequestDTO createRequestDTO(AccessTokenDO accessTokenDO, OAuthAppDO auth
         rowDTOs.add(userTenantDomainDTO);
         rowDTOs.add(resourceDTO);
 
-        for (String scope : accessTokenDO.getScope()) {
+        for (String scope : scopes) {
             RowDTO scopeNameDTO = createRowDTO(scope, XACMLScopeValidatorConstants.SCOPE_ID,
                     XACMLScopeValidatorConstants.SCOPE_CATEGORY);
             rowDTOs.add(scopeNameDTO);
         }
         RequestDTO requestDTO = new RequestDTO();
         requestDTO.setRowDTOs(rowDTOs);
-        return requestDTO;
+
+        RequestElementDTO requestElementDTO = PolicyCreatorUtil.createRequestElementDTO(requestDTO);
+        String request = null;
+        try {
+            request = PolicyBuilder.getInstance().buildRequest(requestElementDTO);
+        } catch (PolicyBuilderException e) {
+            throw new IdentityOAuth2Exception(String.format("Exception occurred when building  XACML request of user " +
+                    "%s.", authenticatedUser.toFullQualifiedUsername()), e);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("XACML scope validation request :\n" + request);
+        }
+        return request;
+    }
+
+    /**
+     * Validates the XACML request using XACML engine with the parameters authApp and authzUser, and returns whether
+     * to permit or not.
+     * @param request XACML request.
+     * @param oAuthAppDO Application.
+     * @param authzUser Fully qualified name of the user.
+     * @return Returns true if the XACML response is permit or NotApplicable. Else returns false.
+     * @throws IdentityOAuth2Exception Exception
+     */
+    private boolean isRequestPermit(String request, OAuthAppDO oAuthAppDO, String authzUser) throws IdentityOAuth2Exception {
+
+        boolean permit = false;
+        try {
+            String responseString = OAuthScopeValidatorDataHolder.getInstance().getEntitlementService().getDecision
```

Comment:
If the line exceeds 120 characters, then move '.getDecision(request);' together to the next line

Reply:
done